ocaml/xapi/pvs_proxy_control.ml

```diff
@@ -123,13 +123,33 @@ let start_proxy ~__context vif proxy =
 		let host = Helpers.get_localhost ~__context in
 		let site = Db.PVS_proxy.get_site ~__context ~self:proxy in
 		let sr, vdi = Xapi_pvs_cache.find_or_create_cache_vdi ~__context ~host ~site in
-		update_site_on_localhost ~__context ~site ~vdi ~starting_proxies:[vif, proxy] ();
+		(try
+			update_site_on_localhost ~__context ~site ~vdi ~starting_proxies:[vif, proxy] ()
+		with e ->
```

Comment:
There is already an exception handler right below here, which should be used for this as well. We don't want `currently_attached` set to `true` if there is a failure.